driver/mloMarginRakningLossHost.hpp

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_view_utils.hpp>
+#include <../test/ford.hpp>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloMarginRankingLossForwardRunHost(const miopenTensorDescriptor_t input1Desc,
+                                           const Tgpu* input1,
+                                           const miopenTensorDescriptor_t input2Desc,
+                                           const Tgpu* input2,
+                                           const miopenTensorDescriptor_t targetDesc,
+                                           const Tgpu* target,
+                                           const miopenTensorDescriptor_t outputDesc,
+                                           Tcheck* output,
+                                           float margin,
+                                           float divisor,
+                                           miopenMarginRakningLossReductionMode_t reduction_mode)
+{
+    tensor_view_t<5> I1_tv = get_inner_expanded_tv<5>(miopen::deref(input1Desc));
+    tensor_view_t<5> I2_tv = get_inner_expanded_tv<5>(miopen::deref(input2Desc));
+    tensor_view_t<5> T_tv  = get_inner_expanded_tv<5>(miopen::deref(targetDesc));
+    tensor_view_t<5> O_tv  = get_inner_expanded_tv<5>(miopen::deref(outputDesc));
+    uint64_t tensor_size   = miopen::deref(targetDesc).GetElementSize();
+    float sum_loss         = 0;
+
+    for(uint64_t gid = 0; gid < tensor_size; ++gid)
```

Comment:
Follow [my script](https://github.com/moreh-dev/MIOpen/blob/0049710197fe76cea9c35ec70dbec59ad08b2ee0/driver/smooth_l1loss_driver.hpp#L65) for shorten and parallelism code.

Reply:
resolved